py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,615 @@
+            'icon': self.icons['temp'],
+
+            'c': round(c_data['temp']),
+            'c_min': round(c_data['temp_min']),
+            'c_max': round(c_data['temp_max']),
+
+            'f': round(f_data['temp']),
+            'f_min': round(f_data['temp_min']),
+            'f_max': round(f_data['temp_max']),
+
+            'k': round(k_data['temp']),
+            'k_min': round(k_data['temp_min']),
+            'k_max': round(k_data['temp_max']),
+        })
+
+    def _format_sunrise(self, wthr):
+        # Get the ISO 8601 time
+        raw_time = wthr.get_sunrise_time('iso')
+        dt = datetime.datetime.strptime(raw_time, "%Y-%m-%d %H:%M:%S+00")
+
+        # Format the sunrise
+        return self.py3.safe_format(self.format_sunrise, {
+            'icon': self.icons['sunrise'],
+            'strftime': dt.strftime(self.format_sunrise_time),
+        })
+
+    def _format_sunset(self, wthr):
+        # Get the ISO 8601 time
+        raw_time = wthr.get_sunset_time('iso')
+        dt = datetime.datetime.strptime(raw_time, "%Y-%m-%d %H:%M:%S+00")
+
+        # Format the sunset
+        return self.py3.safe_format(self.format_sunset, {
+            'icon': self.icons['sunset'],
+            'strftime': dt.strftime(self.format_sunset_time),
+        })
+
+    def _format_dict(self, wthr):
+        return {
+            # Standard options
+            'icon': self._get_icon(wthr),
+            'clouds': self._format_clouds(wthr),
+            'rain': self._format_rain(wthr),
+            'snow': self._format_rain(wthr),
+            'wind': self._format_wind(wthr),
+            'humidity': self._format_humidity(wthr),
+            'pressure': self._format_pressure(wthr),
+            'temp': self._format_temp(wthr),
+            'sunrise': self._format_sunrise(wthr),
+            'sunset': self._format_sunset(wthr),
+
+            # Descriptions
+            'descript': wthr.get_status().title(),
+            'descript_long': wthr.get_detailed_status()
+        }
+
+    def _format(self, wthr, fcsts):
+        # Format all sections
+        today = self._format_dict(wthr)
+
+        # Insert forecasts
+        forecasts = []
+        for day in fcsts:
+            future = self._format_dict(day)
+            forecasts.append(self.py3.safe_format(self.forecast_format, future))
+
+        # Give the final format
+        today['forecast'] = self.forecast_separator.join(forecasts)
+        return self.py3.safe_format(self.format, today)
+
+    def weather(self):
+        self._init()
+
+        # Get weather information
+        coords = self._get_coords()
+        text = ''
+        if (coords is not None):
+            wthr = self._get_weather(coords)
+            fcsts = self._get_forecast(coords)
+
+            text = self._format(wthr, fcsts)
+
+        return {
+            'full_text': text,
+            'cached_until': self.py3.time_in(seconds=self.cache_duration)
+        }
+
+
+if (__name__ == '__main__'):
```

Comment:
Here to ponder if we should add `on_click` right too and disable them (fully/partially) to avoid API updates? Especially when somebody would make massive API updates due to scrolling...